crates/bevy_ecs/src/reflect.rs

```diff
@@ -8,9 +8,23 @@ use crate::{
 };
 use bevy_reflect::{impl_reflect_value, FromType, Reflect, ReflectDeserialize};
 
+/// A runtime type-reflectable component
```

Comment:
Not entirely correct. `ReflectComponent` is a type-erased collection of operations on a given component type. By itself, it doesn't contain any reference to a given component. It's meant to be used as metadata that is fetched `Res<TypeRegistryArc>`.

It can be *used* to get a `dyn Reflect` reference given a `World` reference and `Entity`, but by itself it does not hold nor refer to component data at all.

This isn't too intuitive at all, so an explicit example is probably best here.

Reply:
Ah, that's a very useful explanation! I was muddling through based on my first pass examining the raw code.